usr/sbin/registercloudguest

```diff
@@ -86,11 +86,16 @@ def register_modules(
 
             if prod_reg.returncode:
                 registration_returncode = prod_reg.returncode
-                # Even on error SUSEConnect writes messages to stdout, go figure
+                # Older versions of SUSEConnect wrote error messages to stdout
                 error_message = prod_reg.output
+                if not error_message:
+                    error_message = prod_reg.error
                 if (
                     registration_returncode == 67 and
-                    'registration code' in error_message.lower()
+                    (
+                        'registration code' in error_message.lower() or
+                        'system credentials' in error_message.lower()
+                    )
```

Comment:
I would change this code part to be more future prepared to something like this 

```python
error_indicators = [
    'registration code',
    'system credentials'
]

if (registration_returncode == 67 and error_message.lower() in error_indicators):
    ...
```

It's imho a nicer read, will help you with writing a test and is an easy change if there will be more to check from the error data

Just an idea, thus not relevant for my approval

Reply:
That would not work because the error message will be some longer string and thus it would never match, the `in` test must be from the keywords we are looking for to the error message we receive from the registration command.

Reply:
ok this makes sense. Sorry I did not took into account that the complete error message looks differently.